client/my-sites/post-relative-time-status/index.jsx

```diff
@@ -21,121 +20,200 @@ class PostRelativeTime extends React.PureComponent {
-		const time = moment( includes( [ 'draft', 'pending', 'future' ], status ) ? modified : date );
-		if ( now.diff( time, 'days' ) >= 7 ) {
-			// Like "Mar 15, 2013 6:23 PM" in English locale
-			return time.format( 'lll' );
+			displayedTime = timestamp.calendar( null, {
+				sameElse,
+			} );
 		}
 
-		// Like "3 days ago" in English locale
-		return time.fromNow();
+		// If the content is scheduled to be release within a year, do not display the year at the end
+		return timestamp.diff( now, 'years' ) > 0
+			? displayedTime
+			: displayedTime.replace( timestamp.format( 'Y' ), '' );
 	}
 
 	getTimeText() {
 		const time = this.getTimestamp();
-		if ( ! time ) {
-			return null;
-		}
-
 		return (
 			<span className="post-relative-time-status__time">
-				<Gridicon icon="time" size={ this.props.gridiconSize || 18 } />
-				<time className="post-relative-time-status__time-text" dateTime={ time }>
-					{ this.getDisplayedTimeFromPost( this.props.post ) }
-				</time>
+				{ time && (
+					<>
+						<Gridicon icon="time" size={ this.props.gridiconSize || 18 } />
+						<time className="post-relative-time-status__time-text" dateTime={ time }>
+							{ this.getDisplayedTimeForLabel( true ) }
+						</time>
+					</>
+				) }
 			</span>
 		);
 	}
 
 	getStatusText() {
 		const status = this.props.post.status;
-		let statusClassName = 'post-relative-time-status__status';
-		let statusIcon = 'aside';
-		let statusText;
-
-		if ( this.props.post.sticky ) {
-			statusText = this.props.translate( 'sticky' );
-			statusClassName += ' is-sticky';
-			statusIcon = 'bookmark-outline';
-		} else if ( status === 'pending' ) {
-			statusText = this.props.translate( 'pending review' );
-			statusClassName += ' is-pending';
-		} else if ( status === 'future' ) {
-			const moment = this.props.moment;
-			const now = moment();
-			const scheduledDate = moment( this.props.post.date );
-			// If the content is scheduled to be release within a year, do not display the year at the end
-			const scheduledTime = scheduledDate.calendar( null, {
-				sameElse: this.props.translate( 'll [at] LT', {
-					comment:
-						'll refers to date (eg. 21 Apr) & LT refers to time (eg. 18:00) - "at" is translated',
-				} ),
-			} );
+		const args = {
+			displayedTime: this.getDisplayedTimeForLabel(),
+		};
+		const moment = this.props.moment;
+		const now = moment();
+		const displayOn = Math.abs( now.diff( this.getTimestamp(), 'days' ) ) >= 7;
 
-			const displayScheduleTime =
-				scheduledDate.diff( now, 'years' ) > 0
-					? scheduledTime
-					: scheduledTime.replace( scheduledDate.format( 'Y' ), '' );
+		switch ( status ) {
+			case 'trash':
+				if ( displayOn ) {
+					return this.props.translate( 'trashed on %(displayedTime)s', {
+						comment: '%(displayedTime)s is when a post or page was trashed',
+						args,
+					} );
+				}
+				return this.props.translate( 'trashed %(displayedTime)s', {
+					comment: '%(displayedTime)s is when a post or page was trashed',
```

Comment:
Given absolute/relative dates need to be at a different place for some languages, translators will need to know if this date is relative or absolute. Could you add some extra info in the comments so they can know that we use an absolute date in `trashed on/published on/scheduled for/last modified on [date]` and a relative date in `trashed/published/last modified [date]`?